### PR TITLE
feat: add phisherman plugin

### PR DESCRIPTION
**Pull request by UndiedGamer**

### CODE REVIEW

File: packages/phisherman/src/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+	"extends": "../../../tsconfig.base.json",
+	"compilerOptions": {
+		"rootDir": "./",
+		"outDir": "../dist",
+		"composite": true,
+		"preserveConstEnums": true,
```
Comment (favna) [inline packages/phisherman/src/tsconfig.json:7]:
```suggestion
```

File: packages/phisherman/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+	"extends": "../../tsconfig.base.json",
+	"compilerOptions": {
+		"outDir": "dist",
+		"rootDir": "src",
+		"composite": true,
+		"preserveConstEnums": true,
```
Comment (favna) [inline packages/phisherman/tsconfig.json:7]:
```suggestion
```

File: packages/phisherman/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export * from './lib/Phisherman';
+
+import type { Phisherman, PhishermanOptions } from '.';
```
Comment (favna) [inline packages/phisherman/src/index.ts:3]:
```suggestion
import type { Phisherman, PhishermanOptions } from './lib/Phisherman';
```

File: packages/phisherman/src/register.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { container, Plugin, preGenericsInitialization, SapphireClient } from '@sapphire/framework';
+import type { ClientOptions } from 'discord.js';
+import { Phisherman } from '.';
```
Comment (favna) [inline packages/phisherman/src/register.ts:3]:
```suggestion
import { Phisherman } from './lib/Phisherman';
import './index';
```

The bottom part to ensure that calling `/register` also includes the module augmentations.

File: packages/phisherman/src/lib/Phisherman.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
+import fetch, { Headers } from 'node-fetch';
+import { fromAsync, isErr } from '@sapphire/framework';
+
+export class Phisherman {
+	/**
+	 * The options for phisherman
+	 */
+	public options;
```
Comment (favna) [inline packages/phisherman/src/lib/Phisherman.ts:8]:
```suggestion
	public options: PhishermanOptions;
```

File: packages/phisherman/src/lib/Phisherman.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
+import fetch, { Headers } from 'node-fetch';
```
Comment (favna) [inline packages/phisherman/src/lib/Phisherman.ts:1]: How do you manage to add a dependency on @sapphire/fetch but not use @sapphire/fetch lol

File: packages/phisherman/src/lib/Phisherman.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+export class Phisherman {
+	/**
+	 * The options for phisherman
+	 */
+	public options;
+	public constructor(options: PhishermanOptions | undefined) {
+		this.options = options;
+		void this.checkApiKey(this.options!.apiKey).then((res) => {
+			if (res) throw new Error('Invalid API key');
+		});
```
Comment (favna) [inline packages/phisherman/src/lib/Phisherman.ts:13]: Don't use promises in constructor. That doesn't really work. Instead dedicate this to a method called `init`

File: packages/phisherman/src/lib/Phisherman.ts [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+	 */
+	public async check(domain: string) {
+		const result = await fromAsync(async () => {
+			const check = fetch(`https://api.phisherman.gg/v2/domains/check/${domain}`, {
+				headers: new Headers({
+					'Content-Type': 'application/json',
+					Authorization: `Bearer ${this.options?.apiKey}`
+				})
+			}).then((body) => body.json());
+			return check as Promise<PhishermanReturnType>;
```
Comment (favna) [inline packages/phisherman/src/lib/Phisherman.ts:29]:
```suggestion
			return await fetch<PhishermanReturnType>(`https://api.phisherman.gg/v2/domains/check/${domain}`, {
				headers: {
					'Content-Type': 'application/json',
					Authorization: `Bearer ${this.options?.apiKey}`
				}
			}, FetchResultTypes.JSON);
```

After changing fetch to @sapphire/fetch

**Review by favna — CHANGES_REQUESTED** [review]

File: packages/phisherman/package.json [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+			"import": "./register.mjs",
+			"require": "./register.js"
+		}
+	},
+	"sideEffects": [
+		"./dist/register.js",
+		"./register.js",
+		"./register.mjs"
+	],
+	"homepage": "https://github.com/sapphiredev/plugins/tree/main/packages/editable-commands",
```
Comment (noftaly) [inline packages/phisherman/package.json:26]:
```suggestion
	"homepage": "https://github.com/sapphiredev/plugins/tree/main/packages/phisherman",
```

**Review by noftaly — CHANGES_REQUESTED** [review]

File: packages/phisherman/README.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+[![GitHub](https://img.shields.io/github/license/sapphiredev/plugins)](https://github.com/sapphiredev/plugins/blob/main/LICENSE.md)
+[![codecov](https://codecov.io/gh/sapphiredev/plugins/branch/main/graph/badge.svg?token=QWL8FB16BR)](https://codecov.io/gh/sapphiredev/plugins)
+[![npm bundle size](https://img.shields.io/bundlephobia/min/@sapphire/plugin-phisherman?logo=webpack&style=flat-square)](https://bundlephobia.com/result?p=@sapphire/plugin-phisherman)
+[![npm](https://img.shields.io/npm/v/@sapphire/plugin-phisherman?color=crimson&logo=npm&style=flat-square)](https://www.npmjs.com/package/@sapphire/plugin-phisherman)
+
+</div>
+
+## Description
+
+With plugin-phisherman, you can have out of box integration with [Phisherman](https://phisherman.gg).
```
[the remark this answers is not in the document]
Comment (noftaly) [reply packages/phisherman/README.md:18]: Maybe also add a line or two on what Phisherman is?

**Review by favna — CHANGES_REQUESTED** [review]
Other than the one comment from @noftaly above there's the following to address

- In the description add what phisherman is. I would add this but their site is down at time of writing.
- This should not be a plugin but rather can be moved to the utilities monorepo and be published as `@sapphire/phisherman`. I did some of the heavy lifting for you already by preparing most of the code, you just need to move it and create the PR, then update 
  - `CODEOWNERS`
  - `.npm-deprecaterc.yml`
  - root README (add the badge)
  - Validate the code builds, as I cannot test that until it's in that repo.

File: packages/phisherman/src/lib/Phisherman.ts [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+		const result = await fromAsync(async () => {
+			const check = fetch(`https://api.phisherman.gg/v2/domains/check/verified.test.phisherman.gg`, {
+				headers: new Headers({
+					'Content-Type': 'application/json',
+					Authorization: `Bearer ${key}`
+				})
+			}).then((body) => body.json());
+			return check as Promise<{ success: boolean; message: string }>;
+		});
+		return result.value!.message === 'missing permissions or invalid API key';
```
[the remark this answers is not in the document]
Comment (UndiedGamer) [reply packages/phisherman/src/lib/Phisherman.ts]: They don't give an error code, if they're not authorised it returns
```json
{
"success" : false,
"message": "whatever I was checking there"
}
```

**Pull request closed without merging**